sdk-actors/src/test/java/io/dapr/actors/runtime/ActorStatefulTest.java

```diff
@@ -0,0 +1,661 @@
+    proxy.invokeActorMethod("incrementAndGetCount", 0, int.class).block();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void intTypeWithRuntimeException() {
+    ActorProxy proxy = newActorProxy();
+
+    proxy.invokeActorMethod("getCountButThrowsException", int.class).block();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void actorRuntimeException() {
+    ActorProxy proxy = newActorProxy();
+
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+
+    proxy.invokeActorMethod("forceDuplicateException").block();
+  }
+
+  @Test(expected = IllegalCharsetNameException.class)
+  public void actorMethodException() {
+    ActorProxy proxy = newActorProxy();
+
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+
+    proxy.invokeActorMethod("throwsWithoutSaving").block();
+
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+  }
+
+  @Test
+  public void rollbackChanges() {
+    ActorProxy proxy = newActorProxy();
+
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+
+    // Runs a method that will add one message but fail because tries to add a second one.
+    proxy.invokeActorMethod("forceDuplicateException")
+      .onErrorResume(throwable -> Mono.empty())
+      .block();
+
+    // No message is set
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+  }
+
+  @Test
+  public void partialChanges() {
+    ActorProxy proxy = newActorProxy();
+
+    Assert.assertFalse(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+
+    // Runs a method that will add one message, commit but fail because tries to add a second one.
+    proxy.invokeActorMethod("forcePartialChange")
+      .onErrorResume(throwable -> Mono.empty())
+      .block();
+
+    // Message is set.
+    Assert.assertTrue(proxy.invokeActorMethod("hasMessage", Boolean.class).block());
+
+    // It is first message and not the second due to a save() in the middle but an exception in the end.
+    Assert.assertEquals("first message",
+      proxy.invokeActorMethod("getMessage", String.class).block());
+  }
+
+  private ActorProxy newActorProxy() {
+    ActorId actorId = newActorId();
+
+    // Mock daprClient for ActorProxy only, not for runtime.
+    DaprClient daprClient = mock(DaprClient.class);
+
+    when(daprClient.invokeActorMethod(
+      eq(context.getActorTypeInformation().getName()),
+      eq(actorId.toString()),
+      any(),
+      any()))
+      .thenAnswer(invocationOnMock ->
+        this.manager.invokeMethod(
+          new ActorId(invocationOnMock.getArgument(1, String.class)),
+          invocationOnMock.getArgument(2, String.class),
+          context.getActorSerializer().unwrapMethodResponse(
+            invocationOnMock.getArgument(3, String.class), String.class))
+          .map(s -> {

```

Comment:
Can you intent (add 4 spaces) from .map downwards?  Otherwise it looks like it's lined up with unwrapMethodResponse() and acting on the result of that.

Reply:
2 spaces rather.

Reply:
Nevermind, it is formatted correctly, but there is just a lot packed into these surrounding lines.